### PR TITLE
Homework 2 done

### PR DESCRIPTION
**Pull request by Masahiro21**
Lesson 2 homework

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -3,10 +3,19 @@
...
+
+
+
+#/welcome/ is the URL endpoint, method accepted is GET
+@app.route('/', methods=['GET'])
+def welcome():
+    return "Welcome to the Home Page!"
+
+if __name__ == "__main__":
+    app.run(debug=True)
```
Comment (nobelang78) [inline main.py:21]: Best to specify a port, if not it may default to different port depending on the system

File: Profiles/ProfilesAPI.py [hunk 2]
```diff
@@ -1,7 +1,69 @@
...
+        
+
+@profiles_api.route("/<int:id>", methods = ["DELETE"])
+def delete_profile(id):
+    if request.method == 'DELETE':
+        if id > len(db) - 1:
+            return jsonify({"status":"error","message":"profile doesn't exist"})
+        
+        else:
+            db.remove(id)
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:50]: The remove() method removes the first matching element (which is passed as an argument) from the list.

Removing an index is done like this.
del db[id]

File: Auth/AuthAPI.py [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+
+
+
+@auth_api.route("/register", methods=['POST'])
+def register():
+    if request.method == 'POST':
+        username = request.args.get('username')
+        password = request.args.get('passwordHash')
+        if(username == '') or (password == '') :
+            return jsonify({'status':'error','message':'username or password field is empty'})
```
Comment (nobelang78) [inline Auth/AuthAPI.py:20]: If you want to use the jsonify method from the Flask package. `from Flask import jsonify, ..., ... `

File: Auth/AuthAPI.py [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
 auth_api = Blueprint("auth", __name__)
+
+
+
+@auth_api.route("/register", methods=['POST'])
+def register():
+    if request.method == 'POST':
+        username = request.args.get('username')
+        password = request.args.get('passwordHash')
+        if(username == '') or (password == '') :
```
Comment (nobelang78) [inline Auth/AuthAPI.py:19]: Check if value is None as well

File: Auth/AuthAPI.py [hunk 5]
```diff
@@ -1,7 +1,36 @@
...
+
+@auth_api.route("/login", methods=['POST'])
+def login():
+    if request.method == 'POST':
+        username = request.args.get('username')
+        passwordHash = request.args.get('passwordHash')
+        user_details = {'username':username,'passwordHash':passwordHash}
+        #Upon successful login
+        if user_details in details:
+            token = jwt.econde({'username':username,'passwordHash':passwordHash},current_app.config['SECRET_KEY'],algorithm = 'HS256')
```
Comment (nobelang78) [inline Auth/AuthAPI.py:33]: typo on encode

Comment (nobelang78) [reply Auth/AuthAPI.py:33]: For current_app.config['SECRET_KEY'] to be used, ['SECRET_KEY'] needs to be configured on main.py

**Review by codecodecoder78 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**